liwords-ui/src/chat/players.tsx

```diff
@@ -0,0 +1,294 @@
+import { PlayerAvatar } from '../shared/player_avatar';
+import { moderateUser } from '../mod/moderate';
+import { Form, Input } from 'antd';
+import { UsernameWithContext } from '../shared/usernameWithContext';
+import './playerList.scss';
+import axios from 'axios';
+import { toAPIUrl } from '../api/api';
+import { debounce } from '../utils/debounce';
+import { useMountedState } from '../utils/mounted';
+
+type Props = {
+  defaultChannelType?: string;
+  sendMessage?: (uuid: string, username: string) => void;
+};
+
+type PlayerProps = {
+  className?: string;
+  username?: string;
+  uuid?: string;
+  channel?: string[];
+  fromChat?: boolean;
+  sendMessage?: (uuid: string, username: string) => void;
+};
+
+const Player = React.memo((props: PlayerProps) => {
+  const profile = useBriefProfile(props.uuid);
+
+  const online = props.fromChat || (props.channel && props.channel?.length > 0);
+  let inGame =
+    props.channel &&
+    props.channel.filter((c) => c.includes('chat.game.')).length > 0;
+  let watching =
+    props.channel &&
+    props.channel.filter((c) => c.includes('chat.gametv.')).length > 0;
+  if (!props.username) {
+    return null;
+  }
+  return (
+    <div
+      className={`player-display ${!online ? 'offline' : ''} ${
+        inGame ? 'ingame' : ''
+      } ${props.className ? props.className : ''}`}
+      key={props.uuid}
+    >
+      <PlayerAvatar
+        player={{
+          avatar_url: profile?.getAvatarUrl(),
+          user_id: props.uuid,
+          nickname: props.username,
+        }}
+      />
+      <div>
+        <p className="player-name">
+          <UsernameWithContext
+            username={props.username}
+            userID={props.uuid}
+            moderate={moderateUser}
+            includeFlag
+            omitBlock={props.className === 'friends'}
+            showModTools
+            sendMessage={props.sendMessage}
+          />
+        </p>
+        {inGame || watching ? (
+          <p className="player-activity">
+            {inGame ? 'Playing OMGWords' : 'Watching OMGWords'}
+          </p>
+        ) : null}
+      </div>
+    </div>
+  );
+});
+
+type SearchResponse = {
+  users: Array<Partial<FriendUser>>;
+};
+
+export const Players = React.memo((props: Props) => {
+  const { useState } = useMountedState();
+  const { friends } = useFriendsStoreContext();
+  const { loginState } = useLoginStateStoreContext();
+  const { sendMessage } = props;
+  const { username, loggedIn } = loginState;
+  const [maxHeight, setMaxHeight] = useState<number | undefined>(0);
+  const [searchResults, setSearchResults] = useState<
+    Array<Partial<FriendUser>>
+  >([]);
+  const [searchText, setSearchText] = useState('');
+  const { presences } = usePresenceStoreContext();
+
+  const setHeight = useCallback(() => {
+    const tabPaneHeight = document.getElementById('chat')?.clientHeight;
+    setMaxHeight(tabPaneHeight ? tabPaneHeight - 117 : undefined);
```

Comment:
I would love to see an experiment in that. I don't think you'll be able to have it flex to fill vertically because you can't set a parent height that way. Also, with overflow-y, I think there HAS to be an explicit max height.